src/modal/index.ts

```diff
@@ -0,0 +1,54 @@
+import { alias, tag, utils, Tag } from '@storefront/core';
+
+@alias('modal')
+@tag('gb-modal', require('./index.html'), require('./index.css'))
+class Modal {
+
+  refs: {
+    content: HTMLDivElement,
+  };
+
+  props: Modal.Props = {
+    autoOpen: false,
+  };
+
+  init() {
+    this.set({ visible: this.props.autoOpen });
+  }
+
+  handleOpen = (event: MouseEvent) => {
+    const { body, addEventListener } = utils.WINDOW().document;
+    event.preventDefault();
```

Comment:
I would do these things before anything else (I consider them to be outside of the normal flow of the function)

```js
event.preventDefault();
event.stopPropagation();

const { body, addEventListener } = utils.WINDOW().document;
addEventListener('click', this.close);
this.set({ visible: true });
/* the rest of the function */
```